src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -150,6 +151,42 @@ def get_ops_map(self) -> Dict[int, PauliOperator]:
             ]
         )
 
+    @abstractmethod
+    def get_y_free_equivalent(self):
+        """Return the equivalent of current block but without Y operator."""
+        y_op_indices = list()
+        y_free_block = copy.deepcopy(self)
+
+        # Find Y operators and modify them into X operators
+        for index, operator in enumerate(self.ops_list):
+            if operator == PauliOperator.Y:
+                y_op_indices.append(index)
+                y_free_block.ops_list[index] = PauliOperator.X
+
+        if not y_op_indices:
+            return [y_free_block]
+
+        left_rotations = list()
+        right_rotations = list()
+        if len(y_op_indices) % 2 == 0:
+            # For even numbers of Y operators, add 2 additional pi/4 rotations (one on each side)
+            first_y_operator = y_op_indices.pop(0)
+            new_rotation_ops = [
+                PauliOperator.Z if i == first_y_operator else PauliOperator.I
+                for i in range(self.qubit_num)
+            ]
+            left_rotations.append(PauliRotation.from_list(new_rotation_ops, Fraction(1, 4)))
+            right_rotations.append(PauliRotation.from_list(new_rotation_ops, Fraction(-1, 4)))
```

Comment:
Ok taking a more careful look, we've only shown this to be true as long as we are assuming `self` is a pi/8 rotation.  Because adding the corrective pi/4 rotations is just the math from Litinski's Figure 10. However it might also work in other cases

Y operators need to be removed otherwise for pi/4 rotations and measurement. The way the mat was proved was by adding the corrective terms both on one side as `I = exp(-i*P*pi/4) exp(-i*P*pi/4)` and then commuting one of the terms past the original rotation, which made the Y disappear: `exp(-i*P*phi) = exp(-i*P*pi/4) exp(i*P*pi/4) exp(-i*P*phi) = exp(-i*P*pi/4) exp(-i*Q*phi) exp(i*P*pi/4)` (same thing for the case where two corrective rotations per sideare needed). So I would expect  the same calculation to work for `phi=pi/4` and possibly for measurement since it is subject to the same commutation rules. 

@isolatedinformation I't like you input as well, here

Reply:
Litinski's case is true for `phi=pi/8`. Consider `exp(-i*P*phi) = cos(phi) I - i sin(phi) P`. For `phi=pi/8`, `cos(phi)!=sin(phi)`. So we can effectively rule it out it is just true for that particular angle as it they are not the same

Reply:
WLOG phi=pi/8 then.

Reply:
It remains to show this for measurements

Reply:
@isolatedinformation can you check if it is the same for measurements?

Reply:
I have written the proofs for both general angle Y rotations and Measurements in the proofs goodnotes notebook. It stands true as conjectured.